profiler/src/ProfilerEngine/Datadog.Profiler.Native/StackSamplerLoopManager.cpp

```diff
@@ -472,8 +472,7 @@ bool StackSamplerLoopManager::AllowStackWalk(ManagedThreadInfo* pThreadInfo)
         return false;
     }
 
-    pThreadInfo->AddRef();
-    _pTargetThread = pThreadInfo;
+    _pTargetThread = std::move(pThreadInfo);
```

Comment:
Since pThreadInfo is copied (i.e. AddRef) when passed to this function, does this move forbid the destructor to Release when exiting the function?

Reply:
This move allows to transfer ownership to the `_pThreadInfo` field. So no additional release.